src/lib/util/debug.js

```diff
@@ -24,27 +24,60 @@
  *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  **/
 
+/**
+ * Utility module to handle debugging.
+ * @module util/debug
+ * @see module:util/debug
+ */
+
 'use strict';
 
 const
 	_ = require('lodash'),
 	coreDebug = require('debug'),
-	debugStream = require('debug-stream'),
-
-	addBufferFormatter = (debug) => {
+	debugStream = require('debug-stream');
 
-		coreDebug.formatters.b = (buffer) => {
-			const lines = _.compact(_.split(buffer, '\n'));
-			_.each(_.initial(lines), (value) => {
-				debug(value);
-			});
-			return _.last(lines) || '';
-		};
+/**
+ * Adds the buffer formatter to the debugger.
+ * @instance
+ * @param {*} debug - The debug instance
+ */
+function addBufferFormatter(debug) {
 
+	coreDebug.formatters.b = (buffer) => {
+		const lines = _.compact(_.split(buffer, '\n'));
+		_.each(_.initial(lines), (value) => {
+			debug(value);
+		});
+		return _.last(lines) || '';
 	};
 
+}
+
+function enableLogging(config, namespace) {
+
+	const log = coreDebug(namespace);
+
+	if (!_.get(config, 'silent', false) && _.get(config, 'debug', false)) {
+		coreDebug.enable(namespace);
+	}
+
+	return log;
+
+}
+
+function log(config, namespace, message) {
+	enableLogging(config, namespace)(message);
+}
+
+function stringify(config, namespace, message) {
+	enableLogging(config, namespace)(JSON.stringify(message, undefined, 2));
```

Comment:
2 is magic. Consider `const`-ifying with a descriptive name that tells me what this mysterious numeric argument means (although I now know from googling).